src/script/redmine-upkeep.py

```diff
@@ -524,11 +531,12 @@ def get_filters():
                 REDMINE_STATUS_ID_PENDING_BACKPORT,
                 REDMINE_STATUS_ID_RESOLVED,
             ]
-            yield {
-                f"cf_{REDMINE_CUSTOM_FIELD_ID_PULL_REQUEST_ID}": '>=0',
-                f"cf_{REDMINE_CUSTOM_FIELD_ID_MERGE_COMMIT}": '!*',
-                "status_id": ",".join([str(x) for x in statuses]),
-            }
+            for status in statuses:
+                yield {
+                    f"cf_{REDMINE_CUSTOM_FIELD_ID_PULL_REQUEST_ID}": '>=0',
+                    f"cf_{REDMINE_CUSTOM_FIELD_ID_MERGE_COMMIT}": '!*',
+                    "status_id": str(status),
+                }
 
```

Comment:
Glad that we're no longing hitting the GH API limits! I'm curious: what about that on our side (Redmine)?

Do we have any measure about performance impact from this? With an API key (available to all Redmine users), you can easily trigger complex DB queries.

[CC: @djgalloway]

Does Redmine have any type of quota/throttling mechanism in place? Anyone with an API key and this script might easily trigger a DoS.

Reply:
I asked @akraitman to bump the API limits for the upkeep bot account. I think I observed a server error once but otherwise it seems pretty stable. This script actually self-throttles pretty well because of all the git history commands it executes.

Also, most metadata this script manipulates is now read-only to everyone on our redmine except "Administrators" (the redmine superusers, not the redmine group with the same name) and the bot account. So they can try to run the script but it won't do much (maybe change an issue status).

Reply:
redmine is limiting traffic to different location like /issues /projets but I have excluded the backport bot from any traffic limitation